build.gradle

```diff
@@ -384,6 +384,73 @@ def getCurrentGitBranch() {
+                '-PinitialS3UrbansimInput=`s3 bucket with initial urbansim input starting from double slash. Used as initial data for urbansim.` ' +
+                '-PinitialS3UrbansimOutput=`s3 bucket with initial urbansim output starting from double slash, Used as initial data for first beam run.` ' +
+                '-PinitialSkimsPath=`s3 path to initial skim file. if specified then first beam run will be skipped.`'
+        if (!project.hasProperty('runName')) {
+            throw new GradleException('Please name the run by specifying `runName` argument. e.g; ./gradlew deployPilates -PrunName=sfbay-performance-run' + runExample)
+        }
+        if (!project.hasProperty('startYear')) {
+            throw new GradleException('`startYear` argument required.' + runExample)
+        }
+        if (!project.hasProperty('countOfYears')) {
+            throw new GradleException('`countOfYears` argument required.' + runExample)
+        }
+        if (!project.hasProperty('beamItLen')) {
+            throw new GradleException('`beamItLen` argument required.' + runExample)
+        }
+        if (!project.hasProperty('urbansimItLen')) {
+            throw new GradleException('`urbansimItLen` argument required.' + runExample)
+        }
+        if (!project.hasProperty('initialS3UrbansimInput')) {
+            throw new GradleException('`initialS3UrbansimInput` argument required.' + runExample)
+        }
+    }
+
+    def pload = """{
+  "title": "${project.findProperty('runName')}",
+  "branch": "${project.findProperty('beamBranch') ?: getCurrentGitBranch()}",
+  "commit": "${project.findProperty('beamCommit') ?: 'HEAD'}",
+  "start_year": "${project.findProperty('startYear')}",
+  "count_of_years": "${project.findProperty('countOfYears')}",
+  "beam_it_len": "${project.findProperty('beamItLen')}",
+  "urbansim_it_len": "${project.findProperty('urbansimItLen')}",
+  "initial_urbansim_input": "${project.findProperty('initialS3UrbansimInput')}",
+  "initial_urbansim_output": "${project.findProperty('initialS3UrbansimOutput')}",
+  "pilates_image_version": "${project.findProperty('pilatesImageVersion') ?: 'latest'}",
+  "pilates_image_name": "${project.findProperty('pilatesImageName') ?: 'beammodel/pilates'}",
+  "pilates_scenario_name": "${project.findProperty('pilatesScenarioName') ?: 'pilates'}",
+  "initial_skims_path": "${project.findProperty('initialSkimsPath') ?: ''}",
+  "output_bucket_base_path": "${project.findProperty('outputBucketBasePath') ?: '//pilates-outputs'}",
+  "in_year_output": "${project.findProperty('inYearOutput') ?: 'off'}",
+  "configs": "${project.findProperty('beamConfigs') ?: project.findProperty(getCurrentGitBranch() + '.configs')}",
+  "max_ram": "${maxRAM ?: '140g'}",
+  "storage_size": ${project.findProperty('storageSize')},
+  "batch": "$beamBatch",
+  "instance_type": "${project.findProperty('instanceType') ?: defaultInstanceType}",
```

Comment:
because instanceType is specified in gradle.properties and .. looks like it is `double default` value (=
if it is not specified as parameter it will be read from gradle.properties and only if there no `instanceType` variable even in gradle.properties then `defaultInstanceType` will be used.

by the way, in my copy of gradle.properties there is a `defaultInstanceType=t2.small` line